### ISSUE TITLE
In which order are the results stored in pkl for each test frame?

### ISSUE THREAD

**Issue by selmadeac** [open-mmlab/mmdetection3d#576]
Hi! I don't understand how are the results saved in pkl file after testing. I see a pkl contains a list of dictionaries, each containing information about the objects inside a frame. I suppose the number of dictionaries should be the number of test frames, right? But it is not. There are 7518 frames in kitti test set and only 3769 frames with objects stored in pkl. And if I try to assign first dictionary to frame 000000 from kitti, it is not it. How is the correspondence between frames and detected objects done? How do I know which dictionary is for which frame. I suppose the number of dictionaries is not the same as the number of frames because in some frames there are no objects.

**Comment by THU17cyz** [comment]
Hi, I believe you are using the kitti validation set instead of test set.

**Comment by selmadeac** [comment]
Thank you very much for your explanation! How do I have access to the validation points set ?

**Comment by selmadeac** [comment]
I have a folder in both training and testing folder called velodyne_reduced. But both folders have over 7000> elements inside them.

**Comment by selmadeac** [comment]
I found out. The file indexes are inside ImageSet folder. val.txt. It is clear now. But if I run test.py, I only validate, then if I want to run on the entire test files, I have to write my own code, right? I prefer storing the results in results.pkl even when testing because they are easier to visualize at any time.

**Comment by Wuziyi616** [comment]
Hi, you need to modify the info file you use. You can modify [here](https://github.com/open-mmlab/mmdetection3d/blob/master/configs/_base_/datasets/kitti-3d-3class.py#L131) from `kitti_infos_val.pkl` to `kitti_infos_val.pkl`.

**Comment by THU17cyz** [comment]
Yes of course. It's a part of the official kitti training set. People split that into a smaller training set and val set in order to assess the performance before submitting to the kitti test server. See the kitti data preparation process. congyezhen71 ***@***.*** 签名由 网易邮箱大师 定制 On 05/24/2021 18:31, Selma Deac wrote: Thank you very much for your explanation! How do I have access to the validation points set ? — You are receiving this because you commented. Reply to this email directly, view it on GitHub, or unsubscribe.

**Comment by Wuziyi616** [comment]
Feel free to re-open this issue should you have any further questions :)
